lib/ansible/modules/remote_management/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -0,0 +1,216 @@
+        description: Network share user in the format 'user@domain' or 'domain\\user' if user is
+            part of a domain else 'user'. This option is mandatory for CIFS Network Share.
+    share_pwd:
+        required: False
+        description: Network share user password. This option is mandatory for CIFS Network Share.
+    share_mnt:
+        required: True
+        description: Local mount path of the network share with read-write permission for ansible user.
+            This option is mandatory for Network Share.
+    reboot:
+        required: False
+        description: Whether to reboots after applying the updates or not.
+        default: False
+        type: bool
+    job_wait:
+        required: False
+        description: Whether to wait for job completion or not.
+        type: bool
+        default: True
+    catalog_file_name:
+        required: False
+        description: Catalog file name relative to the I(share_name).
+        type: str
+        default: 'Catalog.xml'
+
+requirements:
+    - "omsdk"
+    - "python >= 2.7.5"
+author: "Rajeev Arakkal (@rajeevarakkal)"
+"""
+
+EXAMPLES = """
+---
+- name: Update firmware from repository on a Network Share
+  dellemc_idrac_firmware:
+       idrac_ip: "192.168.0.1"
+       idrac_user: "user_name"
+       idrac_pwd: "user_pwd"
+       share_name: "192.168.0.0:/share"
+       share_user: "share_user_name"
+       share_pwd: "share_user_pwd"
+       share_mnt: "/mnt/share"
+       reboot: True
+       job_wait: True
+       catalog_file_name: "Catalog.xml"
+"""
+
+RETURN = """
+---
+msg:
+  type: string
+  description: Over all firmware update status.
+  returned: always
+  sample: "Successfully updated the firmware."
+update_status:
+  type: dict
+  description: Firmware Update job and progress details from the iDRAC.
+  returned: success
+  sample:
+    sample: {
+        'InstanceID': 'JID_XXXXXXXXXXXX',
+        'JobState': 'Completed',
+        'Message': 'Job completed successfully.',
+        'MessageId': 'REDXXX',
+        'Name': 'Repository Update',
+        'JobStartTime': 'NA',
+        'Status': 'Success',
+    }
+"""
+
+
+from ansible.module_utils.remote_management.dellemc.dellemc_idrac import iDRACConnection
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from omsdk.sdkcreds import UserCredentials
+    from omsdk.sdkfile import FileOnShare
+    HAS_OMSDK = True
+except ImportError:
+    HAS_OMSDK = False
+
+
+def _validate_catalog_file(catalog_file_name):
+    normilized_file_name = catalog_file_name.lower()
+    if not normilized_file_name:
+        raise ValueError('catalog_file_name should be a non-empty string.')
+    elif not normilized_file_name.endswith("xml"):
+        raise ValueError('catalog_file_name should be an XML file.')
+
+
+def update_firmware(idrac, module):
+    """Update firmware from a network share and return the job details."""
+    msg = {}
+    msg['changed'] = False
+    msg['update_status'] = {}
+
+    try:
```

Comment:
You should probably put this check into a separate function and run it before `with` block https://github.com/ansible/ansible/pull/46675/files#diff-62a00abb6ff3cd2c35f54b8b82436130R205 so that it'll halt execution even before trying to construct that connection object.

Reply:
right.. this input validation can be before target connection